llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -234,76 +209,250 @@ def __init__(
         eval_run_config: Optional[Dict[str, Any]] = None,
     ):
 
+        # Run these during init to fail fast in any of the error cases
         for required in ('save_interval', 'save_folder'):
             if required not in training_params:
                 raise ValueError(f'{required} required for async eval')
 
         self.checkpoint_save_folder = training_params['save_folder']
         self.training_params = training_params
         self.eval_run_config = validate_eval_run_config(eval_run_config)
-        self.interval = validate_interval(interval,
-                                          self.training_params['save_interval'])
-        self.check_interval = create_interval_scheduler(
-            interval,
-            # There is a custom close to ensure that the final checkpoint
-            # (which is the most important) is evaled after it is written
-            include_end_of_training=False,
-        )
-        self.last_checkpoint: Optional[str] = None
 
-        # Run these during init to fail fast in any of the error cases
         self.current_run = self._get_current_run()
         get_eval_parameters(
             parameters=training_params,
             checkpoint='test',
             training_run_name=self.current_run.name,
         )
-        log.info(
-            f'Initialized AsyncEval callback. Will generate runs at interval {interval}'
+
+        # Validate the interval (how often to launch eval runs)
+        self.interval = validate_interval(interval,
+                                          self.training_params['save_interval'])
+
+        # Configures how often to check for new checkpoints
+        self.check_interval = Time(max(self.interval.value // 5, 1),
+                                   self.interval.unit)
+
+        # Keep track of checkpoints by interval that have already been evaled
```

Comment:
As I continue through I think it would be good to rename `interval` throughout when it actually means "the timestamp of this particular checkpoint/eval. Would be much more clear for me to read. lmk if you disagree